pkg/actors/actors.go

```diff
@@ -356,9 +356,10 @@ func (a *actorsRuntime) callRemoteActorWithRetry(
 
 		code := status.Code(err)
 		if code == codes.Unavailable || code == codes.Unauthenticated {
-			_, err = a.grpcConnectionFn(context.TODO(), targetAddress, targetID, a.config.Namespace, false, true, false)
-			if err != nil {
-				return nil, err
+			_, teardown, cerr := a.grpcConnectionFn(context.TODO(), targetAddress, targetID, a.config.Namespace, false, true, false)
+			teardown()
```

Comment:
Should this be deferred?

Reply:
`defer` is not used intentionally.

This line just creates a new connection and does not receive it, so `teardown()` can be called immediately.

Also, since this line is in a `for` loop, `defer` is evaluated after the whole loop ends. All old connections created by each iteration remain until the end of the function.

I referred to this post `#2 — Defer inside a loop`.
https://blog.learngoprogramming.com/gotchas-of-defer-in-go-1-8d070894cb01